app/uk/gov/hmrc/cdsreimbursementclaimfrontend/connectors/UpscanConnector.scala

```diff
@@ -45,12 +45,13 @@ trait UpscanConnector {
     hc: HeaderCarrier
   ): EitherT[Future, Error, HttpResponse]
 
-  def initiate(
+  def initiate[A](
```

Comment:
Is there anyway to constrain the type of `A` so that `initiate` can only be called with a valid type?

Reply:
No it is not possible. Because there is no much common between SupportingEvidenceAnswer which is NonEmptyList of supporting evidences and ScheduledDocumentAnswer which is just single ScheduledDocument object. 
Even putting `implicitNotFound` annotation might be misleading as apart from FileUpload there is also HeaderCarrier implicit in the scope.